.github/workflows/my-ci-pipeline.yml

```diff
@@ -2,9 +2,30 @@ name: Continuous Integration
 on: [push] 
 jobs: 
  build: 
-  name: Build and test 
+  name: build
   runs-on: ubuntu-latest 
   steps: 
   - uses: actions/checkout@v2 
   - run: docker build --target test --tag todo-app:test .
-  - run: docker run --env-file ./.env.test todo-app:test
\ No newline at end of file
+  - run: docker run --env-file ./.env.test todo-app:test
+ deploy:
+  name: deploy
+  needs: build
+  runs-on: ubuntu-latest
+  if: github.event_name == 'push' && github.ref == 'refs/heads/Module8'
+  steps:
+   - uses: actions/checkout@v2
+   - run: docker login -u keithleverton -p ${{secrets.DOCKERPASSWORD}}
+   - run: docker build --target production --tag keithleverton/todo-app:$GITHUB_SHA .
+   - run: docker push keithleverton/todo-app:$GITHUB_SHA
+   - run: curl https://cli-assets.heroku.com/install-ubuntu.sh | sh
```

Comment:
This isn't bad practice by any means, but note that the Heroku CLI is actually pre-installed - GitHub runners come with a mass of pre-installed tools (e.g. for [the latest Ubuntu this is the list](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md) )